Account/FirefoxAccountConfiguration.swift

```diff
@@ -51,12 +51,19 @@ public protocol FirefoxAccountConfiguration {
     var sync15Configuration: Sync15Configuration { get }
 
     var pushConfiguration: PushConfiguration { get }
+
+    var prefs: Prefs? { get }
+
+    init(prefs: Prefs?)
```

Comment:
In what way? We're accepting `Prefs` as part of the initializer now so that we can persist some state.

Reply:
We're only persisting state in very few cases.

In the cases we are persisting state, we should probably generate an immutable configuration from the prefs, rather than passing a mutable configuration.